core/src/main/java/com/datastrato/gravitino/authorization/GroupManager.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.EntityAlreadyExistsException;
+import com.datastrato.gravitino.EntityStore;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.exceptions.GroupAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchEntityException;
+import com.datastrato.gravitino.exceptions.NoSuchGroupException;
+import com.datastrato.gravitino.meta.AuditInfo;
+import com.datastrato.gravitino.meta.CatalogEntity;
+import com.datastrato.gravitino.meta.GroupEntity;
+import com.datastrato.gravitino.storage.IdGenerator;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * GroupManager is used for add, remove and get groups from one metalake. GroupManager doesn't
+ * manage group, just sets up the relationship between the metalake and the group. Metalake is like
+ * a concept of the organization. `AddGroup` means that a group enter an organization.
+ */
+public class GroupManager {
```

Comment:
I think we can merge the User and GroupManager together as "UserGroupManager", WDYT?

Reply:
Done.